src/Sulu/Bundle/AdminBundle/Metadata/SchemaMetadata/PropertyMetadata.php

```diff
@@ -11,49 +11,35 @@
 
 namespace Sulu\Bundle\AdminBundle\Metadata\SchemaMetadata;
 
-class PropertyMetadata
+class PropertyMetadata extends AbstractPropertyMetadata
```

Comment:
@sulu/core-team If I would do that one again, I would probably name it `ConstMetadata`. Initially I thought having one `PropertyMetadata` will be enough, but then we might end up with a huge constructor when adding more stuff to validate. I've already had the `properties` from the `ArrayMetadata` added here as well, but I decided to split these two.

Question is if we should rename this class (I would suggest so), and if we introduce a BC layer (Creating a new class called `ConstMetadata` extending from this one, and deprected the `PropertyMetadata` class).

I would go for the deprecation, and if we do that, I would first get rid of all deprecations that currently have no alternative (if there is no alternative it should not be a deprecation) and configure our [ŧests to fail if we use some of our own deprecated code](https://symfony.com/doc/current/components/phpunit_bridge.html#internal-deprecations).